ocp_utilities/infra.py

```diff
@@ -79,6 +82,45 @@ def assert_pods_failed_or_pending(pods):
         )
 
 
+def assert_nodes_in_healthy_condition(nodes):
+    """
+    Validates nodes are in a healthy condition.
+    None of the following node conditions is True:
+        - DiskPressure
+        - MemoryPressure
+        - PIDPressure
+        - NetworkUnavailable
+
+    Args:
+         nodes(list): List of Node objects
+
+    Raises:
+        NodesNotHealthyConditionError: if one or more nodes condition is other than Ready, i.e
+                                       DiskPressure, MemoryPressure, PIDPressure, NetworkUnavailable
+    """
+    LOGGER.info("Verify all nodes are in a healthy condition")
+    unhealthy_nodes = []
+    for node in nodes:
+        node_conditions = node.instance.status.conditions
+        condition_type = []
+        for condition in node_conditions:
+            if (
+                condition.type != Node.Condition.READY
+                and condition.status == Node.Condition.Status.TRUE
+            ):
+                condition_type.append(condition.type)
+
+        if condition_type:
```

Comment:
Instead of a list, maybe have a dict where the node is the key and the value is list of condition_types
and `unhealthy_nodes` should be all dict that you have

merged_dict = {**dict_one, **dict_two, **dict_three}

Reply:
The recent function, `assert_pods_failed_or_pending` used a tuple. So following the same pattern: And printing the output as:
```
...
! _pytest.outcomes.Exit: Following are nodes with unhealthy condition/s:
        ('[master-0.rhtestc2.lab.upshift.rdu2.redhat.com](http://master-0.rhtestc2.lab.upshift.rdu2.redhat.com/)', 'MemoryPressure', 'DiskPressure', 'PIDPressure')
        ('[master-1.rhtestc2.lab.upshift.rdu2.redhat.com](http://master-1.rhtestc2.lab.upshift.rdu2.redhat.com/)', 'MemoryPressure', 'DiskPressure', 'PIDPressure')
        ('[master-2.rhtestc2.lab.upshift.rdu2.redhat.com](http://master-2.rhtestc2.lab.upshift.rdu2.redhat.com/)', 'MemoryPressure', 'DiskPressure', 'PIDPressure')
        ('[worker-0.rhtestc2.lab.upshift.rdu2.redhat.com](http://worker-0.rhtestc2.lab.upshift.rdu2.redhat.com/)', 'MemoryPressure', 'DiskPressure', 'PIDPressure')
        ('[worker-1.rhtestc2.lab.upshift.rdu2.redhat.com](http://worker-1.rhtestc2.lab.upshift.rdu2.redhat.com/)', 'MemoryPressure', 'DiskPressure', 'PIDPressure')!
```